### PR TITLE
WT-11321 Free fast-truncated pages in checkpoint cleanup (v6.0)

### PR DESCRIPTION
**Pull request by mm-ng**
I have attached a test on the corresponding Jira ticket to demonstrate the issue. I don't plan on merging in the test.

### CODE REVIEW

File: src/btree/bt_sync.c [hunk 1]
```diff
@@ -147,11 +147,16 @@ __sync_ref_obsolete_check(WT_SESSION_IMPL *session, WT_REF *ref)
...
+    /*
+     * Mark the parent page as dirty to remove a fast-deleted child page.
+     *
+     * Ignoring these fast-deleted child pages may result in a disk leak if we downgraded from a
+     * version that writes out fast-delete information. Fast-delete information is ignored in
+     * versions 6.0 and prior, and so the fast-delete operation on these pages is considered
+     * globally visible. Therefore, clean up these pages.
+     */
+    if (ref->state == WT_REF_DELETED)
+        WT_RET(__wt_page_parent_modify_set(session, ref, true));
```
Comment (kommiharibabu) [inline src/btree/bt_sync.c:159]: Usually, the internal page will be dirty when there exists a fast truncated page that is not obsolete to let it clean later. In scenarios when the internal page is evicted with non-obsolete fast truncated pages, marking it dirty here will free those pages.

File: src/btree/bt_sync.c [hunk 2]
```diff
@@ -147,11 +147,16 @@ __sync_ref_obsolete_check(WT_SESSION_IMPL *session, WT_REF *ref)
...
 
     /* Lock the WT_REF. */
     WT_REF_LOCK(session, ref, &previous_state);
```
Comment (kommiharibabu) [inline src/btree/bt_sync.c:162]: Can you add the test to the PR also to let it test in evergreen later?

Comment (mm-ng) [reply src/btree/bt_sync.c:162]: @kommiharibabu I won't be able to add the test in this PR as it is using the Python test/compatibility infrastructure which was implemented in a later branch, but I can create another ticket to add it into develop if required.

Comment (kommiharibabu) [reply src/btree/bt_sync.c:162]: Understood. Thanks for the details.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**